cmd/katalyst-agent/app/agent/context.go

```diff
@@ -109,6 +112,30 @@ func (c *GenericContext) Run(ctx context.Context) {
 	<-ctx.Done()
 }
 
+// customizedPluginManager requires that handlers must be added before
+// it really runs to avoid logging error logs too frequently
+type customizedPluginManager struct {
+	enabled atomic.Bool
+	pluginmanager.PluginManager
+}
+
+// Run successes iff at-least-one handler has been registered
+func (p *customizedPluginManager) Run(sourcesReady config.SourcesReady, stopCh <-chan struct{}) {
+	for range time.Tick(time.Second) {
+		if p.enabled.Load() {
+			p.PluginManager.Run(sourcesReady, stopCh)
+			klog.Infof("started plugin-manager since handlers have been enabled")
```

Comment:
this log seems meaningless since the `PluginManager.Run` function only returns when it stops